### PR TITLE
PIM-7469: add enriched entity icon

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

Small PR to support attribute link to product for enriched entity

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Done
| Added legacy Behats               | NA
| Added acceptance tests            | NA
| Added integration tests           | NA
| Changelog updated                 | NA
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php [hunk 1]
```diff
@@ -46,8 +46,15 @@ public function validate($entity, Constraint $constraint)
 
         foreach ($constraint->properties as $property) {
             $originalValue = $accessor->getValue($originalData, sprintf('[%s]', $property));
+            if (null === $originalValue) {
+                $originalValue = $accessor->getValue($originalData, sprintf('[properties][%s]', $property));
+            }
+
             if (null !== $originalValue) {
                 $newValue = $accessor->getValue($entity, $property);
```
Comment (SamirBoulil) [inline src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php]: You could use a private to get the value in the property or in the "properties".

Comment (juliensnz) [reply src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php]: ok, good idea

File: src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php [hunk 2]
```diff
@@ -58,4 +59,38 @@ public function validate($entity, Constraint $constraint)
...
     }
+
+    /**
+     * @param  PropertyAccessor $accessor
+     * @param  mixed            $originalData
+     * @param  string           $property
+     *
+     * @return mixed
+     */
+    private function getOriginalValue(PropertyAccessor $accessor, $originalData, string $property)
```
Comment (grena) [inline src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php]: return typehint

Comment (juliensnz) [reply src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php]: it's mixed. No idea what it can be

Comment (juliensnz) [reply src/Pim/Component/Catalog/Validator/Constraints/ImmutableValidator.php]: it's mixed. No idea what it can be :)

**Comment by juliensnz** [comment]
@jmleroux you can see the icon in the diff :) https://www.dropbox.com/s/nc8u7xh6u3q6370/Screenshot%202018-07-04%2014.08.25.png?dl=0

**Pull request merged**